source/Launcher/Launcher.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Microsoft.Win32;
+using System;
+using System.Diagnostics;
+using System.Drawing;
+using System.Windows.Forms;
+
+namespace Launcher
+{
+    public partial class BannerlordCoopLauncher : Form
+    {
+        private bool mouseDown;
+        private Point lastLocation;
+        public BannerlordCoopLauncher()
+        {
+            InitializeComponent();
+        }
+
+        /// <summary>
+        /// Check if MB is installed (through the registry) - Only works in x64.
+        /// </summary>
+        /// <returns>Full path where MB is installed</returns>
+        private string GetBannerLordInstalledPath()
+        {
+
+            string registry_key = @"SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall";
+            using (Microsoft.Win32.RegistryKey key = Registry.LocalMachine.OpenSubKey(registry_key))
+            {
+                foreach (string subkey_name in key.GetSubKeyNames())
+                {
+                    using (RegistryKey subkey = key.OpenSubKey(subkey_name))
+                    {
+                        string name = (string)subkey.GetValue("DisplayName");
+                        if (name != null && name.Equals("Mount & Blade II: Bannerlord"))
+                        {
+                            return subkey.GetValue("installlocation").ToString();
+                        }
+                    }
+                }
+            }
+            return null;
+        }
+
+
+        private void TopBar_MouseUp(object sender, MouseEventArgs e)
+        {
+            mouseDown = false;
+        }
+        private void TopBar_MouseDown(object sender, MouseEventArgs e)
+        {
+            mouseDown = true;
+            lastLocation = e.Location;
+        }
+        private void TopBar_MouseMove(object sender, MouseEventArgs e)
+        {
+            if (mouseDown)
+            {
+                this.Location = new Point(
+                    (this.Location.X - lastLocation.X) + e.X, (this.Location.Y - lastLocation.Y) + e.Y);
+
+                this.Update();
+            }
+        }
+
+        private void CloseButton_Click(object sender, EventArgs e)
+        {
+            Close();
+        }
+        /// <summary>
+        /// Start a new process to run the application, then close it. Launch the game in client mode
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void HostGame_Click(object sender, EventArgs e)
```

Comment:
this is pretty identical to the join game click method, lets extract them out to a new method and set the arguments as appropriate in here